daq/dhcp_monitor.py

```diff
@@ -97,5 +97,5 @@ def _dhcp_error(self, e):
         LOGGER.error('DHCP monitor %s error: %s', self.name, e)
         if self.dhcp_log:
             self.dhcp_log.write('Monitor error %s\n' % e)
-        self.cleanup(forget=False)
         self.callback('error', None, exception=e)
+        self.cleanup()
```

Comment:
Are you sure this works in the case where there's a legit DHCP error?  E.g., if the stream is closed first, causing the error?  The purpose of the (forget=False) argument, IIRC, is to handle the case where stream closes, error happens, and then the stream shouldn't be closed again (which would be an error).  Anyway, as long as it works it's all good -- but please check that case.  You can do this by manually (externally to DAQ) killing the DHCP monitor process and see what happens.